packages/apollo-server-plugin-base/src/index.ts

```diff
@@ -66,6 +66,20 @@ export interface ApolloServerPlugin<
 
 export interface GraphQLServerListener {
   serverWillStop?(): ValueOrPromise<void>;
+  // At most one plugin's serverWillStart may return a GraphQLServerListener
+  // with this method. If one does, it is called once on server startup and the
+  // page it returns is served to clients with `accept: text/html` headers. This
+  // is an intentionally simple API; if you want to do something fancy to serve
+  // your UI, you probably should just define a handler in your web framework.
+  renderUIPage?(): UIPage;
+  // An internal-only flag used in the default UI plugin so that it can be
+  // ignored if you install another UI plugin.
+  __internal_installed_implicitly__?: boolean;
```

Comment:
In a similar spirit to my previous thoughts about exposing things like `__testing__` on the interfaces, would it be worth just not putting this here and `@ts-ignore`-ing the small number of places we call it?

Reply:
I don't love ts-ignore, but I came up with a way to be basically typesafe here (using a type guard).